.github/workflows/differential-shellcheck.yml

```diff
@@ -0,0 +1,26 @@
+# Doc: https://github.com/redhat-plumbers-in-action/differential-shellcheck#usage
+---
+
+name: Differential ShellCheck
+on:
+  pull_request:
+    branches: [master]
+
+permissions:
+  contents: read
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Repository checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+
+      - name: Differential ShellCheck
+        uses: redhat-plumbers-in-action/differential-shellcheck@v3
+        with:
+          severity: warning
+          token: ${{ secrets.GITHUB_TOKEN }}
```

Comment:
Since we're not uploading findings as comments anymore, can we drop the token argument? Is it used for anything in read-only mode?

Reply:
My [previous comment was not correct](https://github.com/rbenv/ruby-build/pull/2042#discussion_r969460120). Permissions don't have any effect on Pull-Request review comments. I have tested different configurations of permissions. Sorry about that, It was a new thing for me.

Commenting and uploading of sarif is handled via GitHub bot, and I don't have any control over it. Also, I couldn't find sufficient documentation about how it works or how to configure the bot. The only thing I was able to find was:
- blog post - https://github.blog/changelog/2022-06-02-users-can-view-and-comment-on-code-scanning-alerts-on-the-conversation-tab-in-a-pull-request/
- general documentation of feature - https://docs.github.com/en/code-security/code-scanning/automatically-scanning-your-code-for-vulnerabilities-and-errors/triaging-code-scanning-alerts-in-pull-requests#viewing-an-alert-on-your-pull-request

I have already raised a question on GitHub/community, but so far, no answer:
- https://github.com/community/community/discussions/33035

It is possible to run `differential-shellcheck` without a token, but you lose all SARIF-related features (comments and warnings on the Files changed tab). However, results would still be accessible upon inspecting console logs of the failed job.